src/util/AbstractMemory/Memory.h

```diff
@@ -0,0 +1,114 @@
+  // encoded as a `size_t` in cpp.
+  size_t memoryInBytes_;
+
+ public:
+  // Very simple custom constructor.
+  Memory(const size_t& amountOfMemoryInBytes = 0)
+      : memoryInBytes_{amountOfMemoryInBytes} {}
+
+  // Default assignment operator.
+  Memory& operator=(const Memory&) = default;
+  Memory& operator=(Memory&&) = default;
+
+  // Custom assignment operator, so that we can assign user literals.
+  Memory& operator=(const size_t& amountOfMemoryInBytes);
+  Memory& operator=(size_t&& amountOfMemoryInBytes);
+
+  /*
+  Return the internal memory amount in the wanted memory unit format.
+  For example: If the internal memory amount is 1024 bytes, than `kilobytes()`
+  would return `1.0`.
+  */
+  size_t bytes() const;
+  double kilobytes() const;
+  double megabytes() const;
+  double gigabytes() const;
+  double terabytes() const;
+  double petabytes() const;
+
+  /*
+  Return the string representation of the internal memory amount in the
+  biggest memory unit, that is equal to, or smaller than, the internal memory
+  amount.
+  Example: 1024 bytes would be returned as `"1 KB"`.
+  */
+  std::string asString() const;
+
+  /*
+  @brief Parse the given string and set the internal memory amount to the amount
+  described.
+
+  @param str A string following `./generated/MemoryDefinitionLanguage.g4`. In
+  short: An amount of bytes described via a user defined literal.
+  */
+  void parse(std::string_view str);
+};
+
+// User defined literals for memory units.
+size_t constexpr operator""_Byte(unsigned long long int bytes) { return bytes; }
+
+size_t constexpr operator""_KB(long double kilobytes) {
+  return convertMemoryUnitsToBytes(kilobytes, numberOfBytesPerKB);
+}
+
+size_t constexpr operator""_KB(unsigned long long int kilobytes) {
+  return convertMemoryUnitsToBytes(kilobytes, numberOfBytesPerKB);
+}
+
+size_t constexpr operator""_MB(long double megabytes) {
+  return convertMemoryUnitsToBytes(megabytes, numberOfBytesPerMB);
+}
+
+size_t constexpr operator""_MB(unsigned long long int megabytes) {
+  return convertMemoryUnitsToBytes(megabytes, numberOfBytesPerMB);
+}
+
+size_t constexpr operator""_GB(long double gigabytes) {
+  return convertMemoryUnitsToBytes(gigabytes, numberOfBytesPerGB);
+}
+
+size_t constexpr operator""_GB(unsigned long long int gigabytes) {
+  return convertMemoryUnitsToBytes(gigabytes, numberOfBytesPerGB);
+}
+
+size_t constexpr operator""_TB(long double terabytes) {
+  return convertMemoryUnitsToBytes(terabytes, numberOfBytesPerTB);
+}
+
+size_t constexpr operator""_TB(unsigned long long int terabytes) {
+  return convertMemoryUnitsToBytes(terabytes, numberOfBytesPerTB);
+}
+
+size_t constexpr operator""_PB(long double petabytes) {
+  return convertMemoryUnitsToBytes(petabytes, numberOfBytesPerPB);
+}
+
+size_t constexpr operator""_PB(unsigned long long int petabytes) {
+  return convertMemoryUnitsToBytes(petabytes, numberOfBytesPerPB);
+}
```

Comment:
Those shouldn't return `size_t` but `Memory` directly. 
And they should be in their own namesapce `ad_utility::memory_literals`, s.t. we can use a single `using namespace` to import them.

Reply:
Done, see [f5f9f83](https://github.com/ad-freiburg/qlever/pull/1032/commits/f5f9f839efb6fcc21d1e4310d3b974e985ce8e91).